src/Buttons.jsx

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import Button from './Button';
+
+function Buttons({ onClick }) {
+  return (
+    <p>
+      {[1, 2, 3, 4, 5].map((i) => (
+        <Button keyNumber={i} onClick={onClick} />
```

Comment:
질문이 하나 있습니다!

React 공식문서를 보니 위 Button 엘리먼트에 key 속성을 부여하는 것이 올바른 key 속성 사용법처럼 보입니다.
(key는 위 [1, 2, 3, 4, 5] 배열의 context에서만 의미가 있으므로)

하지만 실제로 <code>{[1, 2, 3, 4, 5].map((i) => \<Button key={i.toString()} onClick={onClick} /\>)}</code> 이렇게 코드를 작성해보면, Button 엘리먼트에서 해당 값을 제대로 읽지 못합니다. (console.log로 출력해보니 undefined로 값이 나옵니다.)

[에러 메세지]
react.development.js:210 Warning: Button: `key` is not a prop.
    Trying to access it will result in `undefined` being returned.
    If you need to access the same value within the child component,
    you should pass it as a different prop.

그래서 대신 위와 같이 keyNumber 속성으로 key값을 전달하여 구현을 했습니다만 이는 옳은 key 사용법처럼 보이지 않기에, 어떻게 이 문제를 해결해야 할까 싶습니다. 도움 좀 부탁드리겠습니다! 😫